simulator/extender/extender.go

```diff
@@ -0,0 +1,146 @@
+
+// New initializes Service.
+func New(client clientset.Interface, extenderCfgs []v1beta2config.Extender, storeReflector storereflector.Reflector) (*Service, error) {
+	extenders, err := createHTTPExtenders(extenderCfgs)
+	if err != nil {
+		return nil, xerrors.Errorf("createHTTPExtenders: %w", err)
+	}
+	store := resultstore.New()
+	// Register the result store of Extenders to the sharedStore.
+	storeReflector.AddResultStore(store, ResultStoreKey)
+	return &Service{
+		client:    client,
+		extenders: extenders,
+		store:     store,
+	}, nil
+}
+
+// Filter returns the result of the specified filter extender
+// and store it.
+func (s *Service) Filter(id int, args extenderv1.ExtenderArgs) (*extenderv1.ExtenderFilterResult, error) {
+	result, err := s.extenders[id].Filter(args)
+	if err != nil {
+		return nil, xerrors.Errorf("call filter of specified HTTPExtender: %w", err)
+	}
+	s.store.AddFilterResult(args, *result, s.extenders[id].Name())
+	return result, nil
+}
+
+// Prioritize returns the result of the specified prioritize extender
+// and store it.
+func (s *Service) Prioritize(id int, args extenderv1.ExtenderArgs) (*extenderv1.HostPriorityList, error) {
+	result, err := s.extenders[id].Prioritize(args)
+	if err != nil {
+		return nil, xerrors.Errorf("call prioritize of specified HTTPExtender: %w", err)
+	}
+	s.store.AddPrioritizeResult(args, *result, s.extenders[id].Name())
+	return result, nil
+}
+
+// Preempt returns the result of the specified preempt extender
+// and store it.
+func (s *Service) Preempt(id int, args extenderv1.ExtenderPreemptionArgs) (*extenderv1.ExtenderPreemptionResult, error) {
+	result, err := s.extenders[id].Preempt(args)
+	if err != nil {
+		return nil, xerrors.Errorf("call preempt of specified HTTPExtender: %w", err)
+	}
+	s.store.AddPreemptResult(args, *result, s.extenders[id].Name())
+	return result, nil
+}
+
+// Bind returns the result of the specified bind extender
+// and store it.
+func (s *Service) Bind(id int, args extenderv1.ExtenderBindingArgs) (*extenderv1.ExtenderBindingResult, error) {
+	result, err := s.extenders[id].Bind(args)
+	if err != nil {
+		return nil, xerrors.Errorf("call bind of specified HTTPExtender: %w", err)
+	}
+	s.store.AddBindResult(args, *result, s.extenders[id].Name())
+	return result, nil
+}
+
+// OverrideExtendersCfgToSimulator rewrites the config to the extenders requests to go through the simulator server.
+func OverrideExtendersCfgToSimulator(cfg *v1beta2config.KubeSchedulerConfiguration, simulatorPort int) {
+	for i := range cfg.Extenders {
+		// i will be the extender's index. That index is specified by request param as `id`.
+		cfg.Extenders[i].EnableHTTPS = false
+		cfg.Extenders[i].TLSConfig = nil
+		// TODO: change these logic to allow for detailed settings.
+		// Currently, we need to change these URL if the simulator uses HTTPS or your own domain.
+		cfg.Extenders[i].URLPrefix = "http://localhost:" + strconv.Itoa(simulatorPort) + "/api/v1/extender/"
```

Comment:
Could someone give me advice on a better method to get the simulator's config?
In current, our simulator doesn't have an Objects that consolidate  configuration of the simulator's settings into one.